.github/workflows/release.yml

```diff
@@ -17,22 +17,22 @@ jobs:
   goreleaser:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
Is there a reason we're using hashes here instead of the action versions? Usually we would just do `actions/checkout@v4`, etc.

Reply:
It's copied/pasted from https://github.com/hashicorp/terraform-provider-scaffolding-framework/blob/main/.github/workflows/release.yml is why. Whether we want to or not, I don't really have a strong opinion about.

Reply:
I am not really sure on that.
The current file in the codebase uses hashes https://github.com/ansible/terraform-provider-aap/blob/main/.github/workflows/release.yml
and also one from [terraform-provider-scaffolding-framework](https://github.com/hashicorp/terraform-provider-scaffolding-framework) uses hashes https://github.com/hashicorp/terraform-provider-scaffolding-framework/blob/main/.github/workflows/release.yml
so continued with the same

Reply:
Pinning the major version will automatically use minor and patch updates without us having to update it, which I am in favor of.